frameworks/AutoWEKA/exec.py

```diff
@@ -78,19 +78,37 @@ def run(dataset: Dataset, config: TaskConfig):
     probabilities_labels = dataset.target.values
     if not os.path.exists(weka_file):
         raise NoResultError("AutoWEKA failed producing any prediction.")
+
+    collected_labels = [None] * len(probabilities_labels)
+
+    def extract_pred(pred):
+        idx, val = pred.split(':')
+        idx = int(idx)
+        collected_labels[idx-1] = val
+        return val
+
     with open(weka_file, 'r') as weka_file:
         probabilities = []
         predictions = []
         truth = []
         for line in weka_file.readlines()[1:-1]:
             inst, actual, predicted, error, *distribution = line.split(',')
             pred_probabilities = [pred_probability.replace('*', '').replace('\n', '') for pred_probability in distribution]
-            _, pred = predicted.split(':')
-            _, tru = actual.split(':')
+            pred = extract_pred(predicted)
+            tru = extract_pred(actual)
             probabilities.append(pred_probabilities)
             predictions.append(pred)
             truth.append(tru)
 
+    collab = set(collected_labels)
+    unused_labels = [l for l in probabilities_labels if l not in collab]
+    if len(unused_labels) > 1:
```

Comment:
What unused label are we expecting (that makes it `>1` instead of `>0`)?

Reply:
we only collect labels from predictions and truth values so for a multiclass problem (let's say 10 classes), we can imagine one fold for which we have only 7 distinct values in the target feature (and same, more or less, in the predictions), in which case we won't see all values.
If missing only 1, then we know it for sure, but not if missing 2 or more.

Reply:
> If missing only 1, then we know which one it is for sure

That was the part I missed. I should have stopped to think 😅